capabilities/writetarget/write_target.go

```diff
@@ -0,0 +1,479 @@
+//nolint:gosec,revive // disable G115,revive
+package writetarget
+
+import (
+	"context"
+	"encoding/binary"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"time"
+
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/trace"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities"
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities/consensus/ocr3/types"
+	"github.com/smartcontractkit/chainlink-common/pkg/logger"
+	"github.com/smartcontractkit/chainlink-common/pkg/services"
+	commontypes "github.com/smartcontractkit/chainlink-common/pkg/types"
+
+	"github.com/smartcontractkit/chainlink-evm/capabilities/writetarget/report/platform"
+	"github.com/smartcontractkit/chainlink-framework/capabilities/writetarget/beholder/monitor"
+	"github.com/smartcontractkit/chainlink-framework/capabilities/writetarget/retry"
+
+	wt "github.com/smartcontractkit/chainlink-framework/capabilities/writetarget/beholder/pb/platform"
+)
+
+var (
+	_ capabilities.TargetCapability = &writeTarget{}
+)
+
+type TransactionStatus uint8
+
+// new chain agnostic transmission state types
+const (
+	TransmissionStateNotAttempted TransactionStatus = iota
+	TransmissionStateSucceeded
+	TransmissionStateFailed // retry
+	TransmissionStateFatal  // don't retry
+)
+
+// alter TransmissionState to reference specific types rather than just
+// success bool
+type TransmissionState struct {
+	Status      TransactionStatus
+	Transmitter string
+	Err         error
+}
+
+type TargetStrategy interface {
+	// QueryTransmissionState defines how the report should be queried
+	// via ChainReader, and how resulting errors should be classified.
+	QueryTransmissionState(ctx context.Context, reportID uint16, request capabilities.CapabilityRequest) (*TransmissionState, error)
+	// TransmitReport constructs the tx to transmit the report, and defines
+	// any specific handling for sending the report via ChainWriter.
+	TransmitReport(ctx context.Context, report []byte, reportContext []byte, signatures [][]byte, request capabilities.CapabilityRequest) (string, error)
+}
+
+var (
+	_ capabilities.TargetCapability = &writeTarget{}
+)
+
+// chain-agnostic consts
+const (
+	CapabilityName = "write"
+
+	// Input keys
+	// Is this key chain agnostic?
+	KeySignedReport = "signed_report"
+)
+
+type writeTarget struct {
+	capabilities.CapabilityInfo
+
+	config    Config
+	chainInfo monitor.ChainInfo
+
+	lggr logger.Logger
+	// Local beholder client, also hosting the protobuf emitter
+	beholder *monitor.BeholderClient
+
+	cs               commontypes.ChainService
+	cr               commontypes.ContractReader
+	cw               commontypes.ContractWriter
```

Comment:
Wait, why do we still need cr/cw if we're now injecting a TargetStrategy interface to abstract required interactions with the chain. This is now confusing, which interface is used?

I suggest we clean this up and only use a single interface.

Reply:
We still need CW to [check the transaction status](https://github.com/smartcontractkit/chainlink-framework/blob/0647c811e8e34635171517e64571650c59402d6a/capabilities/writetarget/write_target.go#L380) later, but good point we no longer need CR in the generalize wt, only in the target strategy.